configuration/device-tree.md

```diff
@@ -332,10 +310,85 @@ Note that a property is assigned the value `false` by not defining it. Boolean p
+```
+local_mac_address = [aa bb cc dd ee ff];
+```
+
+#### 2.2.5: Parameters with multiple targets
+<a name="part2.2.5"></a>
+
+There are some situations where it is convenient to be able to set the same value in multiple locations within the Device Tree. Rather than the ungainly approach of creating multiple parameters, it is possible to add multiple targets to a single parameter by concatenating them, like this:
+
+```
+    __overrides__ {
+        gpiopin = <&w1>,"gpios:4",
+                  <&w1_pins>,"brcm,pins:0";
+        ...
+    };
+```
+
+(example taken from the `w1-gpio` overlay)
+
+Note that it is even possible to target properties of different types with a single parameter. You could reasonably connect an "enable" parameter to a `status` string, cells containing zero or one, and a proper boolean property.
+
+#### 2.2.6: Literal assignments
+<a name="part2.2.6"></a>
+
+As seen in <a href="part2.2.5">2.2.5</a>, the DT parameter mechanism allows multiple targets to be patched from the same parameter, but the utility is limited by the fact that the same value has to be written to all locations (except for format conversion and the negation available from inverted booleans). The addition of embedded literal assignments allows a parameter to write arbitrary values, regardless of the parameter value supplied by the user.
+
+Assignments appear at the end of a declaration, and are indicated by a `=`:
+
+```
+str_val  = <&target>,"strprop=value";              // 1
+int_val  = <&target>,"intprop:0=42                 // 2
+int_val2 = <&target>,"intprop:0=",<42>;            // 3
+bytes    = <&target>,"bytestr[=b8:27:eb:01:23:45"; // 4
+```
+Lines 1, 2 and 4 are fairly obvious, but line 3 is more interesting because the value appears as an integer (cell) value. The DT compiler evaluates integer expressions at compile time, which might be convenient (particularly if macro values are used), but the cell can also contain a reference to a label:
+```
+// Force an LED to use a GPIO on the internal GPIO controller.
+exp_led = <&led1>,"gpios:0=",<&gpio>,
+          <&led1>,"gpios:4";
+```
+When the overlay is applied, the label will be resolved against the base DTB in the usual way. Note that it is a good idea to split multi-part parameters over multiple lines like this to make them easier to read - something that becomes more necessary with the addition of cell value assignments like this.
+
+Bear in mind that parameters do nothing unless they are applied - a default value in a lookup table is ignored unless the parameter name is used without assigning a value.
+
+#### 2.2.7: Lookup tables
+<a name="part2.2.7"></a>
+
+Lookup tables allow parameter input values to be transformed before they are used. They act as associative arrays, rather like switch/case statements:
+```
+phonetic = <&node>,"letter{a=alpha,b=bravo,c=charlie,d,e,='tango uniform'}";
+bus      = <&fragment>,"target:0{0=",<&i2c0>,"1=",<&i2c1>,"}";
```

Comment:
Perhaps I'm misunderstanding what this is doing, but should the `,"1=",` above be `,",1=",` ?